Steffi dev2

steffilazerte opened the pull request:
Edits for BASSr round 2

### To Do
- [X] Users can supply a non-spatial data frame which will be converted to sf with a crs and coords. 
  - *Only part of 'entry' functions: `full_BASS_run()`, `draw_random_samples()`, `calculate_benefit()`*
- [X] Rename attr_sf to something more intuitive
  - *Currently using `land_raw` (for `clean_land_cover()`) and `land_hex` (for all else) and to be explicit as to what the dataset is. But my first thought was to use `df_sf` (data frame / sf) which is more generic... This is **very** easy to change!*
- [X] Add arguments for setting seeds using `withr::with_seed()`
- [x] Adjust sampling in `run_grts_on_BASS()` (stratum, oversamples, no list, mindis/maxtry, etc.)
- [x] Deprecate functions
  - [x] `noGRTS_BASS_run()`
  - [x] `subsample_grts_and_calc_benefit()`
  - [x] `getresults_BASS()`
  - [x] Full deprecate other functions (i.e. comment out code so we don't need to maintain it to best practices)
- [x] Small
  - [x] Habitat column names should be "area"
  - [x] Explain `omit_flag` in `full_BASS_flag()`
- [x] Internal docs
- [x] Final tests
- [x] All `TODO`
- [x] Missing input checks

steffilazerte commented:
And just a couple of questions/things to consider:

- [ ] Quiet and verbose arguments - Check the places where messages can be suppressed (i.e. search `!quiet`) - Are these appropriate?
- [ ] Should the order of coordinates be latitude/longitude (the way we say it, "lat/lons"), or longitude/latitude (the way many/most spatial functions in R want it, "x,y")
- [ ] `calc_sun()` - are the signs on times to sunrise correct? Right now, a recording that happens 15 min before sunrise has a ttsr of -15... is that correct, or should it be 15?
- [ ] Should the metadata always have a consistent sort order (i.e should it always be based on path? `add_site()` currently sorts by joining variables first)
- `clean_metadata()
  - [ ] `aru_type` is currently extracted without an option for changing the pattern - is that okay?
  - [ ] We don't currently have an option for `aru_model` - do we need one?
- [ ] `clean_gps()` - we don't currently extract serial nubmers or firmware versions - do we need to (or is that part of the log extraction?)

steffilazerte commented:
I think I'm done! Just a couple of final questions/comments here as well:

- See comments in *italics* above
- One last TODO - Create test for `create_study_area()`, but I thought that function was still going to change so I've left it for now
- `run_grts_on_BASS()` is a bit complex but allows a variety of different ways of passing the data. See the BASSr Vignette and examples and let me know what you think
- Looks like `full_BASS_run()` can have n_samples == 0... but how should calculate_benefit() handle this?
- `draw_random_samples()` has `...` but I can't see that it's really ever used?
  - I added mindis and mintry as explicit arguments, does that remove the need for this?

dhope commented:
One thing I noticed was that with update to `dplyr` I'm running into rename_with in clean_land_cover. I can get it to work from changing this [code ](https://github.com/dhope/BASSr/blob/a6e017cb681265fbc91c05815359ad70491abafa/R/Clean4Bass.R#L44-L47) to the following:

`dplyr::rename_with(
    land_raw,
    .fn = ~lc_rename(nms = .x, pattern = pattern, append = append),
    .cols = dplyr::matches(pattern)
    )`


I can make the changes after merging if you're done with the PR, I just didn't want to push into your branch or put the main ahead of your changes as it causes conflicts.

steffilazerte commented:
Hi @dhope, I fixed the outstanding issues. I added an issue for the comment you had regarding future extensions, and opened an issue so we don't forget to check in about the `...` usage (I went with my recommendation for now, we can revisit later as needed). 

I also commented out deprecated/defunct functions so I could run a proper Package Check. The only notes/issues created are with the functions that are slated for the next round and with some of the Vignettes.

I'll send an 'official' email once I've finished the final tweaks on `ARUtools` tomorrow.

dhope commented on R/BASS_studyArea_analysis.R:
```diff
@@ -1,96 +1,224 @@
+                          n_over = n_os,
+                          n_base = n_strata,
+                          stratum_var = stratum_name,
+                          DesignID = "sample",
```
we will want to push DesignID to the `...`. I think for `aux_var` we want users to generate and specify their own columns to use here. That doesn't seem possible with the `...`, so perhaps need to include another variable with a default of "inclpr".

dhope replied:
I'd just want to make sure doing so won't break other places in the code if "inclpr" is needed to be the selection column.

dhope commented on .Rbuildignore:
```diff
@@ -12,4 +12,20 @@
+
+
+# Temp omit for checking
+# ^R/Add_points.R$
```
Not sure what best procedure is here. Should I uncomment these before doing a release or leave as is? Some look like they can stay ignored, but others are still useful.

dhope reviewed (changes requested):
Nice work. I've added a couple comments in the code. Just need to be sure about Rbuildignore and the changes to DesignID and aux_var in grts before merging. Not critical, just don't want them to get lost.
